### PR TITLE
fix(zitadelctl): implementation for commands in crd mode

### PR DESCRIPTION
**Pull request by stebenz**

### CODE REVIEW

File: operator/zitadel/kinds/orb/reconcile.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package orb
 import (
 	"github.com/caos/orbos/mntr"
```
Comment (eliobischof) [inline operator/zitadel/kinds/orb/reconcile.go:4]:
```suggestion
	"fmt"

	"github.com/caos/orbos/mntr"
```

File: operator/zitadel/kinds/orb/reconcile.go [hunk 2]
```diff
@@ -14,36 +15,45 @@ func Reconcile(
...
-		}
-
-		imageRegistry := spec.CustomImageRegistry
-		if imageRegistry == "" {
-			imageRegistry = "ghcr.io"
-		}
+			if spec.Version == "" {
+				err := errors.New("No version provided for self-reconciling")
+				recMonitor.Error(err)
+				return err
```
Comment (eliobischof) [inline operator/zitadel/kinds/orb/reconcile.go:28]:
```suggestion
				return errors.New("no version provided for self-reconciling")
```

File: operator/zitadel/kinds/orb/reconcile.go [hunk 3]
```diff
@@ -14,36 +15,45 @@ func Reconcile(
...
+					recMonitor.Error(errors.Wrap(err, "Failed to deploy zitadel-operator into k8s-cluster"))
+					return err
+				}
+				recMonitor.Info("Applied zitadel-operator")
+			}
+			return nil
+		}, func(k8sClient kubernetes2.ClientInt) error {
+			if err := kubernetes.DestroyZitadelOperator(monitor, labels.MustForAPI(labels.NoopOperator("zitadel-operator"), "zitadel", "v0"), k8sClient, gitops); err != nil {
+				monitor.Error(errors.Wrap(err, "Failed to destroy zitadel-operator in k8s-cluster"))
 				return err
```
Comment (eliobischof) [inline operator/zitadel/kinds/orb/reconcile.go:54]:
```suggestion
				return fmt.Errorf("failed to destroy zitadel-operator in k8s-cluster: %w", err)
```

File: pkg/databases/backup.go [hunk 4]
```diff
@@ -19,6 +20,28 @@ func InstantBackup(
 		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/backup.go:20]:
```suggestion
```

File: pkg/databases/backup.go [hunk 5]
```diff
@@ -19,6 +20,28 @@ func InstantBackup(
...
+}
+
+func CrdInstantBackup(
+	monitor mntr.Monitor,
+	k8sClient kubernetes.ClientInt,
+	name string,
+) error {
+	desired, err := database.ReadCrd(k8sClient)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/backup.go:33]:
```suggestion
```

File: pkg/databases/backuplist.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	monitor mntr.Monitor,
+	gitClient *git.Client,
+	k8sClient kubernetes.ClientInt,
+) (
+	[]string,
+	error,
+) {
+	desired, err := gitClient.ReadTree(git.DatabaseFile)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/backuplist.go:22]:
```suggestion
```

File: operator/zitadel/kinds/orb/reconcile.go (hunk 2 above)
Comment (eliobischof) [inline operator/zitadel/kinds/orb/reconcile.go:28]:
```suggestion
				return errors.New("no version provided for self-reconciling")
```

File: operator/zitadel/kinds/orb/reconcile.go (hunk 3 above)
Comment (eliobischof) [inline operator/zitadel/kinds/orb/reconcile.go:54]:
```suggestion
				return fmt.Errorf("failed to destroy zitadel-operator in k8s-cluster: %w", err)
```

File: pkg/databases/backup.go (hunk 4 above)
Comment (eliobischof) [inline pkg/databases/backup.go:20]:
```suggestion
```

File: pkg/databases/backup.go (hunk 5 above)
Comment (eliobischof) [inline pkg/databases/backup.go:33]:
```suggestion
```

File: pkg/databases/backuplist.go (hunk 6 above)
Comment (eliobischof) [inline pkg/databases/backuplist.go:22]:
```suggestion
```

File: pkg/databases/backuplist.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+func CrdListBackups(
+	monitor mntr.Monitor,
+	k8sClient kubernetes.ClientInt,
+) (
+	[]string,
+	error,
+) {
+	desired, err := database.ReadCrd(k8sClient)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/backuplist.go:38]:
```suggestion
```

File: pkg/databases/backuplist.go [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	monitor mntr.Monitor,
+	k8sClient kubernetes.ClientInt,
+	desired *tree.Tree,
+) (
+	[]string,
+	error,
+) {
+	backups, err := orbdb.BackupListFunc()(monitor, k8sClient, desired)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/backuplist.go:55]:
```suggestion
```

File: pkg/databases/clear.go [hunk 9]
```diff
@@ -5,21 +5,48 @@ import (
...
+func GitOpsClear(
 	monitor mntr.Monitor,
 	k8sClient kubernetes.ClientInt,
 	gitClient *git.Client,
 	databases []string,
+	users []string,
 ) error {
 	desired, err := gitClient.ReadTree(git.DatabaseFile)
 	if err != nil {
 		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/clear.go:22]:
```suggestion
```

File: pkg/databases/clear.go [hunk 10]
```diff
@@ -40,5 +67,6 @@ func Clear(
...
```
Comment (eliobischof) [inline pkg/databases/clear.go:67]:
```suggestion
```

File: pkg/databases/restore.go [hunk 11]
```diff
@@ -21,6 +22,30 @@ func Restore(
 		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/restore.go:22]:
```suggestion
```

File: pkg/databases/restore.go [hunk 12]
```diff
@@ -21,6 +22,30 @@ func Restore(
...
+
+func CrdRestore(
+	monitor mntr.Monitor,
+	k8sClient kubernetes.ClientInt,
+	name string,
+	databases []string,
+) error {
+	desired, err := database.ReadCrd(k8sClient)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/databases/restore.go:36]:
```suggestion
```

File: pkg/zitadel/migration.go [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+
+//Take care! to use this function you have to include migration files into the binary
+func CrdMigrations(
+	monitor mntr.Monitor,
+	k8sClient *kubernetes.Client,
+	version *string,
+) error {
+	desired, err := zitadel.ReadCrd(k8sClient)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/zitadel/migration.go:21]:
```suggestion
```

File: pkg/zitadel/migration.go [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+func GitOpsMigrations(
+	monitor mntr.Monitor,
+	orbCfg *orbconfig.Orb,
+	gitClient *git.Client,
+	k8sClient *kubernetes.Client,
+	version *string,
+) error {
+	desired, err := gitClient.ReadTree(git.ZitadelFile)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/zitadel/migration.go:38]:
```suggestion
```

File: pkg/zitadel/zitadel.go [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+)
+
+func CrdScaleDown(
+	monitor mntr.Monitor,
+	k8sClient *kubernetes.Client,
+	version *string,
+) error {
+	desired, err := zitadel.ReadCrd(k8sClient)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/zitadel/zitadel.go:20]:
```suggestion
```

File: pkg/zitadel/zitadel.go [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+func GitOpsScaleDown(
+	monitor mntr.Monitor,
+	orbCfg *orbconfig.Orb,
+	gitClient *git.Client,
+	k8sClient *kubernetes.Client,
+	version *string,
+) error {
+	desired, err := gitClient.ReadTree(git.ZitadelFile)
+	if err != nil {
+		monitor.Error(err)
```
Comment (eliobischof) [inline pkg/zitadel/zitadel.go:36]:
```suggestion
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

**Comment by fforootd** [comment]
is this still needed?

**Review by eliobischof — APPROVED** [review]

**Review by eliobischof — APPROVED** [review]

**Pull request merged**